app/controllers/posts_controller.rb

```diff
@@ -228,6 +228,11 @@ def close
         render json: { status: 'failed', message: 'Invalid input for other post.' }, status: :bad_request
         return
       end
+      
+      if other = @post
```

Comment:
**error**: you want `other == @post` here - i.e. equality check, rather than assignment